src/unknown.rs

```diff
@@ -0,0 +1,50 @@
+use std::fs::File;
+use std::io;
+use std::path::Path;
+
+static ERROR_MESSAGE: &str = "Unknown Architecture";
```

Comment:
Could we make this error message a bit more precise? e.g., "same-file is not supported on this platform."